library/std/src/io/buffered/bufwriter.rs

```diff
@@ -150,7 +155,11 @@ impl<W: Write> BufWriter<W> {
         impl Drop for BufGuard<'_> {
             fn drop(&mut self) {
                 if self.written > 0 {
-                    self.buffer.drain(..self.written);
+                    if self.done() {
+                        self.buffer.clear();
+                    } else {
```

Comment:
I'm surprised this makes a difference.

Reply:
If it did help, it was negligible. Removed!